app/src/main/java/com/sdp13epfl2021/projmag/database/impl/firebase/FirebaseCommentsDatabase.kt

```diff
@@ -0,0 +1,104 @@
+package com.sdp13epfl2021.projmag.database.impl.firebase
+
+import androidx.appcompat.app.AppCompatActivity
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+import com.sdp13epfl2021.projmag.database.interfaces.CommentsDatabase
+import com.sdp13epfl2021.projmag.database.interfaces.ProjectId
+import com.sdp13epfl2021.projmag.database.interfaces.UserdataDatabase
+import com.sdp13epfl2021.projmag.model.*
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class FirebaseCommentsDatabase @Inject constructor(
+    private val firestore: FirebaseFirestore,
+    private val auth: FirebaseAuth,
+    private val userdataDatabase: UserdataDatabase
+) : CommentsDatabase,  AppCompatActivity() {
+
+    companion object {
+        /**
+         *  The field containing all the comments to all projects
+         */
+        const val PROJECT_COMMENTS = "project-comments"
+    }
+
+    override fun addCommentToProjectComments(
+        message: Message,
+        projectId: ProjectId,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        getCommentsOfProject(projectId, { oldMessages ->
+            val newMessages = oldMessages.toMutableList()
+            newMessages += message
+            val messagesStrings = newMessages.map {message -> message.toMapString()}
+            val docData = hashMapOf(
+                "comments" to messagesStrings
+            )
+            firestore.collection(PROJECT_COMMENTS).document(projectId).set(
+                docData
+            )
+                .addOnSuccessListener{  onSuccess() }
+                .addOnFailureListener { exception -> onFailure(exception) }
+
```

Comment:
It will be easier to read with this indentation : 
```suggestion
            firestore
                .collection(PROJECT_COMMENTS)
                .document(projectId)
                .set(docData)
                .addOnSuccessListener{ onSuccess() }
                .addOnFailureListener(onFailure)
```